### ISSUE TITLE
Debugger stuck with red dump

### ISSUE THREAD

**Issue by justanotheranonymoususer** [x64dbg/x64dbg#2781]
### Operating System

Windows 11

### x64dbg Version

Sep 18 2021

### Describe the issue

Somehow the data I was looking at became invalid (was freed?), which made the dump red. Then the process exited, and the dump stayed red. Then I reloaded the executable, and now I can't get rid of the red without having to restart the debugger.

![image](https://user-images.githubusercontent.com/5781692/139515656-613f6659-47d5-4df2-b005-0b7a3233c366.png)

### Steps to reproduce

1. 
2. 
3.

**Comment by mrexodia** [comment]
The red happens when you double click on data in the reference view (to try
and persuade you to look there).

It would be nice to know how to reproduce it, because the best I can do now
is kinda force disable it when you change the dump address or something.

**Comment by dalerkd** [comment]
Yes, I also encountered this situation a few months ago. It's very uncomfortable because I happen to use multiple memory windows to work.

The difference is that I met it in the X32 version.

If I have no memory errors, I seem to solve this problem by 'breaking away' and then 'closing'?

**Comment by justanotheranonymoususer** [comment]
I started seeing it more lately. Here's a repro. See how if I trigger an animation while another animation is active, it can become stuck in red.


https://user-images.githubusercontent.com/5781692/213740638-37a246b4-7ad9-4990-893e-630373462c65.mp4

**Comment by Steesha** [comment]
I can reproduce a similar "red dump stays red" issue with a deterministic UI path.
  ### Steps to reproduce

  1. Start debugging any x64 process.
  2. Open/focus the Dump view.
  3. Click/select any byte/value in the Dump view.
  4. The status bar/log area at the bottom shows a message like:

  ```text
  module.exe: 0000000000000000 -> 0000000000000000 (0x00000001 bytes)
```
  5. Click the address link in that message.
  6. The Dump view jumps to that address and flashes red 3 times, which appears intentional.
  7. Quickly click the same kind of address link several times, around 5 times or more.
  ### Expected behavior

  The Dump view should flash red temporarily and always return to the normal background color.

  ### Actual behavior

  After clicking the address link repeatedly/quickly, the Dump view can remain red permanently until x64dbg is RESTARTED.

  ### Source observation

  This looks related to the existing dump attention flicker implementation.

  LogView::handleLink() executes dump <addr> for non-code addresses and emits getDumpAttention().

  CPUMultiDump::getDumpAttention() calls CPUDump::getAttention().

  CPUDump::getAttention() starts a new BackgroundFlickerThread.

  BackgroundFlickerThread::run() stores:

  QColor oldColor = background;

  and then alternates background between BackgroundFlickerColor and oldColor.

  If multiple flicker threads overlap, a later thread can start while background is already the red flicker color. In that case its oldColor becomes red, so it restores the dump background to
  red at the end.

  This may explain why rapid repeated clicks can leave the Dump view permanently red.

  Related source:

  - https://github.com/x64dbg/x64dbg/blob/development/src/gui/Src/Gui/LogView.cpp
  - https://github.com/x64dbg/x64dbg/blob/development/src/gui/Src/Gui/CPUMultiDump.cpp
  - https://github.com/x64dbg/x64dbg/blob/development/src/gui/Src/Gui/CPUDump.cpp
  - https://github.com/x64dbg/x64dbg/blob/development/src/gui/Src/Utils/BackgroundFlickerThread.cpp

  ### Environment

  - OS: Windows 11 64bit, Build 26200.
  - x64dbg version: latest snapshot / Apr 20 2026.
  - Architecture: x64.

### About
This bug has been with me for many years, and I hope the developers will fix it as soon as possible!
